data/adapters/gersbach_E2G_perturb_seq_adapter.py

```diff
@@ -0,0 +1,135 @@
+# 0	1.846713305	0.998	0.994	0	NFE2L1-12	ENSG00000087460	ENSG00000082641	chr17	48047370	48048871
+# 0	4.714580706	0.944	0.474	0	EOMES-8	ENSG00000030582	ENSG00000163508	chr3	27721997	27723498
+# 5e-324	2.422421422	0.997	0.966	1.80833e-319	EOMES-8	ENSG00000122862	ENSG00000163508	chr3	27721997	27723498
+# 1.70888203873918e-305	2.353451553	0.904	0.498	6.25467914998928e-301	KLF2-11	ENSG00000095794	ENSG00000127528	chr19	16323826	16325327
+
+
+class GersbachE2GPerturbseq:
+
+    ALLOWED_LABELS = [
+        'genomic_element',
+        'genomic_element_gene'
+    ]
+    SOURCE = 'IGVF'
+
+    def __init__(self, filepath, label, source_url, writer: Optional[Writer] = None, **kwargs):
+        if label not in GersbachE2GPerturbseq.ALLOWED_LABELS:
+            raise ValueError('Ivalid label. Allowed values: ' +
+                             ','.join(GersbachE2GPerturbseq.ALLOWED_LABELS))
+        self.data_file = filepath
+        self.label = label
+        self.source_url = source_url
+        self.file_accession = source_url.split('/')[-2]
+        self.dataset = label
+        self.type = 'edge'
+        if (self.label == 'genomic_element'):
+            self.type = 'node'
+        self.writer = writer
+        self.gene_validator = GeneValidator()
+        self.files_filesets = FileFileSet(
+            self.file_accession, writer=None, label='igvf_file_fileset')
+
+    def process_file(self):
+        self.writer.open()
+
+        file_set_props, _, _ = self.files_filesets.query_fileset_files_props_igvf(
+            self.file_accession, replace=False)
+        simple_sample_summaries = file_set_props['simple_sample_summaries']
+        biosample_term = file_set_props['samples']
+        treatments_term_ids = file_set_props['treatments_term_ids']
+        method = file_set_props['method']
+        genomic_element_to_element_id = {}
+        with gzip.open(self.data_file, 'rt') as data_file:
+            reader = csv.reader(data_file, delimiter='\t')
+            next(reader)
+            for row in reader:
+                p_val = float(row[0])
+                avg_log2FC = float(row[1])
+                pct_1 = float(row[2])
+                pct_2 = float(row[3])
+                p_val_adj = float(row[4])
+                target_gene = row[6]
+                if not self.gene_validator.validate(target_gene):
+                    raise ValueError(
+                        f'Promoted gene: {target_gene} is not a valid gene.')
+                intended_target_name = row[7]
+                if not self.gene_validator.validate(intended_target_name):
+                    raise ValueError(
+                        f'Promoted gene: {intended_target_name} is not a valid gene.')
+                intended_target_chr = row[8]
+                intended_target_start = row[9]
+                intended_target_end = row[10]
+                element_id = build_regulatory_region_id(
```

Comment:
can add a check `if element_coordinates in genomic_element_to_element_id ` and skip `build_regulatory_region_id`, but if it's a small file doesn't matter too much

Reply:
Good idea, I'll add that.